internal/providers/nvidia_bluefield/nvidia_bluefield.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2025 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nvidia_bluefield
+
+import (
+	"os"
+
+	"github.com/coreos/ignition/v2/config/v3_6_experimental/types"
+	"github.com/coreos/ignition/v2/internal/platform"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
+
+	"github.com/coreos/vcontext/report"
+)
+
+const (
+	bootfifoPath = "/sys/bus/platform/devices/MLNXBF04:00/bootfifo"
```

Comment:
As discussed over in https://github.com/coreos/ignition/issues/2148#issuecomment-3491975984 even though there is only one value in the [list upstream](https://github.com/torvalds/linux/blob/6146a0f1dfae5d37442a9ddcba012add260bceb0/drivers/platform/mellanox/mlxbf-bootctl.c#L954) it is still a list. We should make this a list as well and add big comment here with the context linking to the upstream source code.

This implies changes to the code below to (I guess) iterate over this list.

Reply:
> and add big comment here with the context linking to the upstream source code.

We need that big comment and link still.